File: app/src/main/kotlin/org/example/Prueba.kt
package org.example
// Importa las clases necesarias
import com.google.ai.client.generativeai.GenerativeModel
import com.google.ai.client.generativeai.type.GenerateContentResponse
import com.google.ai.client.generativeai.Chat

import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    // 1. Configura el modelo con tu clave de API
    val API_KEY = "AIzaSyAX6g_w_VeRAECmeQqDZEbbwfIxRZ-HYbA"
    val generativeModel = GenerativeModel(
        modelName = "gemini-1.5-flash-latest",
        apiKey = API_KEY
    )

    println("¡Hola! Soy un chatbot. Escribe 'salir' para terminar la conversación.")

    // 2. Inicia la conversación con el modelo
    val chat = generativeModel.startChat()

    // 3. Bucle para interactuar con el usuario
    while (true) {
        print("> Tú: ")
        val userPrompt = readlnOrNull()
        if (userPrompt.equals("salir", ignoreCase = true)) {
            break
        }

        // 4. Envía el mensaje y obtén la respuesta
        val response = chat.sendMessage(userPrompt)

        // 5. Imprime la respuesta del modelo
        val botResponse = response.text
        println("> Bot: $botResponse")
    }
}

File: app/src/main/kotlin/org/example/prueba.py
import google.generativeai as genai
import os

# 1. Configuración de la API Key
# Pega tu clave de la API en la siguiente línea.
# Para obtenerla, ve a: https://aistudio.google.com/app/apikey
genai.configure(api_key="AIzaSyBPSzWgyEnENenleqOD5K5hqWxnn_Bs5AQ")

# 2. Seleccionamos el modelo y creamos una sesión de chat
model = genai.GenerativeModel('gemini-1.5-flash')
chat = model.start_chat(history=[])

print("Hola, soy Gemini. ¡Puedes hacerme todas las preguntas que quieras!")
print("Para terminar, simplemente escribe 'salir'.")

# 3. Bucle para hacer múltiples preguntas
while True:
    # Solicitamos la pregunta al usuario
    pregunta = input("\nTu pregunta: ")

    # Si el usuario escribe 'salir', terminamos el bucle
    if pregunta.lower() == 'salir':
        print("¡Hasta la próxima!")
        break

    # 4. Enviamos la pregunta al chat y mostramos la respuesta
    try:
        # Usamos `send_message` para que el modelo recuerde el contexto
        respuesta = chat.send_message(pregunta)
        print("\nGemini:", respuesta.text)
    except Exception as e:
        print(f"Ocurrió un error: {e}")
        print("Asegúrate de que tu API Key sea válida.")